Release notes for XLT 7.3.0 (#208)
Xceptance/xlt-documentation: content/en/xlt/release-notes/7_3_x.md

---
title: 7.3.x
linkTitle: 7.3.x

weight: 69
type: docs

date: 2023-10-27

description: >
    JS engine off by default, detailed status reporting removed, report generator improvements

sitemap:
  changefreq: never
  priority: 0.1
---

## XLT 7.3.0

See [here](https://github.com/Xceptance/XLT/milestone/29?closed=1) for a complete list of all changes.

### XLT Framework

#### Truly disable the JavaScript engine when not needed

Load test scenarios typically use HtmlUnit and its Web client to load, evaluate, and interact with HTML pages. The Web client is also capable of executing JavaScript code found in a page. In load tests, JavaScript execution is typically disabled to conserve CPU resources. However, the JavaScript *engine* has always been initialized in the Web client.

The Web client has been changed to *not* initialize the JavaScript engine by default. Since the engine is heavy, this reduces CPU and memory usage.

> Whether the engine is to be enabled and whether JavaScript code is executed in the page can be controlled by these properties in the configuration of your test suite:

Reviewer: ```suggestion
Whether the JavaScript engine is to be enabled and whether JavaScript code is executed in the page can be controlled by these properties in the configuration of your test suite:
```

Author: Done.

```
# Whether the Web client supports JavaScript at all (the main switch).
com.xceptance.xlt.javaScriptEngineEnabled = false

# Whether JavaScript code is executed right from the start.
com.xceptance.xlt.javaScriptEnabled = false
```

> If you need JavaScript support in your test scenarios, be sure to add the `com.xceptance.xlt.javaScriptEngineEnabled` property and set both properties to `true`. Similarly, if you don't need JavaScript, set both properties to `false.`

Reviewer: ```suggestion
If you need JavaScript support in your test scenarios, be sure to add both properties stated above and set them to `true`. Similarly, if you don't need JavaScript, set both properties to `false.`
```

Author: Done.

> If you only want to run JavaScript code for a portion of your test scenario, you would need to enable the JavaScript engine, but you can disable JavaScript execution initially. You would then *programmatically* enable and disable the execution of JavaScript code in your test scenario as needed.

Reviewer: ```suggestion
If you want to run JavaScript code only for a portion of your test scenario(s), you would need to enable the JavaScript engine while disabling JavaScript execution initially. You could then *programmatically* enable and disable the execution of JavaScript code in your test scenario as necessary.
```

Author: Done with modifications.

### Load Testing

#### Detailed status reporting removed from Master Controller

> The Master Controller no longer supports listing the status of each individual test user when printing the load test status to the console. Printing the aggregated status per scenario is now the only option. This reduction has allowed us to drastically reduce the amount of data that needs to be transferred from the Agent Controllers to the Master Controller every five seconds. Note that backward compatibility with Agent Controllers from a previous version of XLT 7.x is maintained.

Reviewer: ```suggestion
The Master Controller no longer supports listing the status of each individual test user when printing the load test status to the console. Printing the aggregated status per scenario is now the only option. 
This reduction has allowed us to drastically reduce the amount of data that needs to be transferred from the Agent Controllers to the Master Controller every five seconds. Note that backward compatibility with Agent Controllers running on a previous version of XLT 7.x is maintained.
```

Author: Done.

#### Report generator enhanced

> The Report Generator has been further enhanced to run faster and use fewer resources. This includes optimizations when parsing CSV files and executing request processing rules (also known as merge rules).

Reviewer: ```suggestion
The Report Generator has been tuned in terms of runtime performance and resource consumption. This includes optimizations when parsing CSV files and executing request processing rules (also known as merge rules).
```

Author: Kept original text,

#### List of supported data centers updated

XLT ships with `ec2_admin` and `gce_admin`, our tools for managing machine instances in the Amazon and Google clouds. These tools now support recently opened data centers (regions).

{{% note notitle %}}

> In AWS, not all regions are immediately available to your account, but only after you opt in. `ec2_admin` filters out regions that are not enabled for your account. So if you miss a data center in `ec2_admin`, you may need to opt in first.

Reviewer: ```suggestion
In AWS, not all regions are immediately available to your account, but only after you opted in. `ec2_admin` filters out regions automatically that are not enabled for your account. So if you miss a data center in `ec2_admin`, you may need to opt in first.
```

Author: Done with modifications.

{{% /note %}}

#### Improved "infinity" cells in comparison reports

> Fixed an issue with "infinity" cells in comparison report tables. The infinity symbol is now easier to recognize and has the correct background shading when the mouse is over it.

Reviewer: ```suggestion
Fixed an issue with "infinity" table cells in comparison reports. The infinity symbol is now easier to recognize and has the correct background shading when the mouse is over it.
```
